gapis/resolve/dependencygraph/depgraph.go

```diff
@@ -0,0 +1,300 @@
+package dependencygraph
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/app/benchmark"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/database"
+)
+
+var footprintBuildCounter = benchmark.GlobalCounters.Duration("footprint.build")
+
+// TODO: Remove this ad-hoc command index type and its logic for comparing once
+// the api.CmdID is replaced with []uint64 like type.
+type FullCommandIndex []uint64
+
+func (fci FullCommandIndex) LessThan(fci2 FullCommandIndex) bool {
```

Comment:
Probably should be removed as we now have SubCmdIdx. However, here the command index does include the CmdID of the `vkQueueSubmit`, but SubCmdIdx usually does not.